### ISSUE TITLE
"gogs dump" should take a destination argument

### ISSUE THREAD

**Issue by majewsky** [gogits/gogs#3069]
```
$ sudo -u gogs -g gogs /usr/share/gogs/gogs dump --config /srv/gogs/conf/app.ini
2016/05/06 18:02:03 Creating tmp work dir: /tmp/gogs-dump-187221175
2016/05/06 18:02:03 Dumping local repositories.../srv/gogs/repo
2016/05/06 18:02:47 Fail to dump local repositories: write /tmp/gogs-dump-187221175/gogs-repo.zip: no space left on device
```

`/tmp` is usually a tmpfs on modern distributions and thus limited to 50% the size of the RAM, which in my case is just 0.5 GiB. A `--output/-o` argument should be allowed to specify a different output directory.

**Comment by bkcsoft** [comment]
what you're asking for is a `temp directory argument`, `--output/-o` would be where `gogs-dump-<date>.zip` ends up :smile:

**Comment by msoedov** [comment]
Sorry a bit late response. I agree with @bkcsoft comments on the PR that `--tempdir/-t` would be more accurate.

**Comment by Unknwon** [comment]
This is claimed to be fixed by merging #3086, please help test on `develop` branch.
